src/Optimizer/Transformer/PreloadHeroImage.php

```diff
@@ -521,14 +529,22 @@ private function generateImg(HeroImage $heroImage, Document $document)
             }
         }
 
+        foreach (self::ATTRIBUTES_TO_INLINE as $attribute) {
+            if ($element->hasAttribute($attribute)) {
+                $value = $element->getAttribute($attribute);
+                $style = empty($value) ? '' : "{$attribute}:{$element->getAttribute($attribute)}";
+                $imgElement->addInlineStyle($style);
```

Comment:
Since this can throw an exception, shouldn't the each transformer's `transform` call be wrapped in a `try`/`catch` block in `TransformationEngine::optimizeDom()`:

https://github.com/ampproject/amp-toolbox-php/blob/c7ccb98b6bbd96ed5c5546302f272c70b03116e6/src/Optimizer/TransformationEngine.php#L62-L67

The caught `$exception` can be added to `$errors`. 

I recall we discussed this earlier today. The `MaxCssByteCountExceeded` exception would be classed as a catchable exception which doesn't necessitate abandoning optimization entirely. Or rather, perhaps this one specific call to `$imgElement->addInlineStyle($style);` should be wrapped in a `try {…} catch ( MaxCssByteCountExceeded $exception ) {…}` to then re-throw a catchable `TransformerAbortedException`.

Reply:
I add try/catch clauses around the `addInlineStyle()` calls. I don't there's any point right now in catching exceptions at the transformer level.